frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -0,0 +1,485 @@
+import { Link } from 'react-router-dom'
+import {
+  getFolderContentList,
+  getWorkspaceDetail,
+  getFileContent,
+  getFolderDetail,
+  putFileIsDeleted,
+  getWorkspaceContentList
+} from '../action.async'
+import Carousel from '../component/Carousel.jsx'
+import { removeExtensionOfFilename, debug, DIRECTION } from '../helper.js'
+import ReactImageLightbox, { LightboxRotation } from '../Lightbox.js'
+import 'react-image-lightbox/style.css'
+import Fullscreen from 'react-full-screen'
+import classnames from 'classnames'
+
+const qs = require('query-string')
+
+class Gallery extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      appName: 'gallery',
+      isVisible: true,
+      config: props.data ? props.data.config : debug.config,
+      loggedUser: props.data ? props.data.loggedUser : debug.loggedUser,
+      content: props.data ? props.data.content : debug.content,
+      breadcrumbsList: [],
+      appMounted: false,
+      folderId: qs.parse(props.data.config.history.location.search).folder_ids,
+      imagesPreviews: [],
+      fileCurrentPage: 1,
+      fileName: 'unknown',
+      fileSelected: 0,
+      autoPlay: null,
+      fullscreen: false,
+      displayPopupDelete: false
+    }
+
+    // i18n has been init, add resources from frontend
+    addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
+    i18n.changeLanguage(this.state.loggedUser.lang)
+
+    this.lightboxRotation = new LightboxRotation()
+
+    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  }
+
+  customEventReducer = ({ detail: { type, data } }) => { // action: { type: '', data: {} }
+    const { state } = this
+
+    switch (type) {
+      case CUSTOM_EVENT.SHOW_APP(state.config.slug):
+        console.log('%c<Gallery> Custom event', 'color: #28a745', type, data)
+        if (data.config.appConfig.workspaceId !== state.config.appConfig.workspaceId || qs.parse(data.config.history.location.search).folder_ids !== state.folderId) {
+          this.setState({
+            config: data.config,
+            folderId: qs.parse(data.config.history.location.search).folder_ids
+          })
+        }
+        break
+      case CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE:
+        console.log('%c<Gallery> Custom event', 'color: #28a745', type, data)
+        this.setState(prev => ({
+          loggedUser: {
+            ...prev.loggedUser,
+            lang: data
+          }
+        }))
+        i18n.changeLanguage(data)
+        this.buildBreadcrumbs()
+        break
+      default:
+        break
+    }
+  }
+
+  async componentDidMount () {
+    const { state } = this
+
+    console.log('%c<Gallery> did mount', `color: ${state.config.hexcolor}`)
+
+    if (state.folderId) await this.loadFolderDetail(state.config.appConfig.workspaceId, state.folderId)
+    await this.loadGalleryList(state.config.appConfig.workspaceId, state.folderId)
```

Comment:
these await doesn't seem to be necessary. Check if they are and if not, remove them